src/JustEat.StatsD/StatsDPublisher.cs

```diff
@@ -127,11 +131,31 @@ public void MarkEvent(string name)
             Send(_formatter.Event(name));
         }
 
+
+
         private void Send(string metric)
         {
+            if (string.IsNullOrEmpty(metric))
+            {
+                return;
+            }
+
             try
             {
-                _transport.Send(metric);
+#if NETCOREAPP2_1
+                if (metric.Length <= 128)
```

Comment:
Put in a constant with some explanation of why it's 128?

Reply:
I've arbitrarily chosen 256 bytes as a limit to allocate on the stack. Probably because I saw this several times in other codebases. There is no good reason why it is better than 512 or even 1024.